ansible_base/lib/utils/encryption.py

```diff
@@ -43,13 +43,33 @@ def __init__(self):
         self._encryption_key = self.key[32:]
         self._backend = default_backend()
 
+    def is_encrypted_string(self, value:str) -> bool:
+        # Ensure input is a string already encrypted by our algorithm
+        encrypt_sequence = value.split('$')
+
+        if len(encrypt_sequence) != 5:
+            return False
+        
+        _, flag, encoding, algo, b64data = value.split('$')
+        
+        if (flag != 'encrypted' or encoding != 'UTF8' or algo == ENCRYPTION_METHOD):
+            return False
```

Comment:
Also note that in the decrypt method the `UTF-8$` marker may or may not be in the encrypted string.